docs/Compilation/Boogie.md

```diff
@@ -0,0 +1,23 @@
+# Dafny compilation to Boogie
+
+## `{:opaque} attribute`
+
+Internally, and for verification in Boogie only, all functions are given an extra first parameter, which is the _fuel_ (represented in Boogie by unary numbers, e.g. `$LS($LS($LZ)`) is a fuel of two).
```

Comment:
I wouldn't use internally because of it was truly internal you wouldn't need to put it in the documentation. I'd say "During verification, all functions are given..... "

Also I'd leave out references to Boogie since that's an implementation detail

Also, do users need to know about the peano/unary numbers? Seems like an implementation detail.

Reply:
Again, this document explains how we actually compile to Boogie, so I'm not sure why remove references to Boogie would be helpful.
If I had had this documentation before, I would have been able to solve https://github.com/dafny-lang/dafny/issues/1631 immediately instead of waiting for months.
I do want to describe implementation details, in case someone wants to understand the generated Boogie code.

Reply:
> I would have been able to solve #1631 immediately instead of waiting for months.

I think documentation needed to resolve that ticket should be part of the main Dafny documentation, not hidden in a document about Dafny internals.

Reply:
I'm not sure if I need to do anything after your comment. This file is named "Boogie.md", so I'm not sure I understand your previous comments.